isisd/isis_zebra.c

```diff
@@ -736,6 +596,10 @@ void isis_zebra_init(struct thread_master *master, int instance)
 
 void isis_zebra_stop(void)
 {
+	if (zclient_sync->sock > 0) {
```

Comment:
hmm - I don't see `zclient_sync` being freed anywhere; don't we need to do zclient_stop() and zclient_free()?

Reply:
My concerns is that I'm not sure that zclient_stop() and zclient_free() apply to synchronous socket. Can you tell me if we could use these function to a synchronous client or not? If it applies, of course I'll replace this part with a call to zclient_stop() and 'zclient_free()`, if not I'll just add a free statement.

Reply:
it looks like zclient_new() allocates memory, so it should be freed? I do see that ldp doesn't seem to clean up its "zclient_sync" - @rwestphal is that ... ok? shouldn't ldp (lde.c) clean up the second zclient session object?

Reply:
I replace this part by a call to `zclient_stop(zclient_sync)` and `zclient_free(zclient_sync)` and test it. It is working fine.